src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -352,8 +353,9 @@ public override async ValueTask<bool> HandleRpcAsync(ClientPlayer sender, Client
 
                     Rpc44SetRole.Deserialize(reader, out var role);
                     PlayerInfo.RoleType = role;
+                    _hasRoleSet = true;
 
-                    if (Game.GameState == GameStates.Starting)
+                    if (Game.GameState == GameStates.Starting && Game.Players.All(clientPlayer => clientPlayer.Character is InnerPlayerControl { _hasRoleSet: true }))
```

Comment:
This is quite dirty because it assumes every `IInnerPlayerControl` is an `InnerPlayerControl`.
Do we want to have `HasRoleSet` on `IInnerPlayerControl`? Or maybe make `IInnerPlayerControl.RoleType` nullable?